src/nccl_ofi_rdma.c

```diff
@@ -1428,7 +1428,18 @@ static inline int process_completions(struct fi_cq_tagged_entry *cq_entry,
 				return ncclInternalError;
 			}
 
-			if (IS_CONN_RESP_MSG_TYPE(cq_entry[comp_idx].tag) && (comp_flags & FI_RECV)) {
+			if (IS_CONN_MSG_TYPE(cq_entry[comp_idx].tag) && (comp_flags & FI_SEND)) {
+				assert(req->comm->type == NCCL_NET_OFI_SEND_COMM);
+				nccl_net_ofi_rdma_send_comm_t *s_comm =
+					(nccl_net_ofi_rdma_send_comm_t *)req->comm;
+				assert(req == s_comm->send_conn_req);
+				/* Release connect message request */
+				req->free(req, false);
+				req = NULL;
+				s_comm->send_conn_req = NULL;
+				__sync_synchronize();
```

Comment:
why is the sync_synchronize() needed here?

Reply:
I was handling it in the same way we handle the connected flag: https://github.com/aws/aws-ofi-nccl/blob/02dc354bab2c145f8cf4c91c6fb12ffb9f7df16d/src/nccl_ofi_rdma.c#L2225-L2230

I think the thinking is that we are polling the `connect_msg_sent` variable on a possibly different thread from the one that sets it, so we need to make sure the polling thread (`blocked_send_close`) doesn't free resources before the setting thread (`process_completions`) finishes using it.

Reply:
Ok; that's wrong, too.  But open a SIM to fix this and let's move on.